### ISSUE TITLE
Scss string interpolation as custom property default #4999 

### ISSUE THREAD

**Issue by pascalduez** [csstools/stylelint-value-no-unknown-custom-properties#12]
> Clearly describe the bug

Using an SCSS string interpolation for a custom property `var()` function default produce a `CSSSyntaxError`.

> Which rule, if any, is the bug related to?

`csstools/value-no-unknown-custom-properties`

> What code is needed to reproduce the bug?

```css
$color: #fff;

.test {
  color: var(--color, #{$color});
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "plugins": ["stylelint-value-no-unknown-custom-properties"],
  "rules": {
    "csstools/value-no-unknown-custom-properties": [
      true,
      {
        "importFrom": "src/styles/global/variables.css"
      }
    ]
  }
}
```

> Which version of stylelint are you using?

`13.7.2`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint --syntax scss 'src/**/*.scss'`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to SCSS string interpolation.

> What did you expect to happen?

No error.

> What actually happened (e.g. what warnings or errors did you get)?

The following error were flagged:

```shell
$ stylelint --syntax scss 'src/**/*.scss'

<input css 1>
 1:11  ✖  Unknown word   CssSyntaxError
```

**Comment by kmuncie** [comment]
Is there any chance of this getting sorted out?

**Comment by oleksii-boiko** [comment]
Same problem

**Comment by iliatapazukk** [comment]
As I understand it, this is the problem of the linter itself, which does not support interpolation

**Comment by pascalduez** [comment]
> As I understand it, this is the problem of the linter itself, which does not support interpolation

I wouldn't say so. Remove the plugin from config, all fine.

**Comment by jameschensmith** [comment]
I was able to figure out what the exact issue is. I've listed the full error message below for reference. The problem is [this line](https://github.com/csstools/stylelint-value-no-unknown-custom-properties/blob/82887fd23f3e9ab37a32a674192e9612a501497b/src/lib/validate-decl.js#L8). What is happening is that `postcss-values-parser` is being used to parse the PostCSS declaration value. From my understanding, the values aren't normally parsed by PostCSS, which is why this is done here. But, `postcss-values-parser` doesn't understand the SCSS interpolation, and so, it triggers an unknown word error when it reaches that.

Stylelint is also using this library for [`custom-property-pattern`](https://github.com/stylelint/stylelint/blob/99ec404e3b9dd5dc0020e00920ef87fbae64aa83/lib/rules/custom-property-pattern/index.js), but they have some checks that I need to dig deeper into which may be the reason it's not throwing an error for them. Until I find that out, the only solution I can think of to address this error is to capture the CSS custom properties through regex rather than this parsing library. Maybe someone else can chime in with a better idea.

<details>
<summary>Full error message</summary>

```
CssSyntaxError: <css input>:1:10: Unknown word
    at Input.error ([...]/stylelint-value-no-unknown-custom-properties/node_modules/postcss-values-parser/node_modules/postcss/lib/input.js:128:16)
    at ValuesParser.unknownWord ([...]/stylelint-value-no-unknown-custom-properties/node_modules/postcss-values-parser/node_modules/postcss/lib/parser.js:561:22)
    at ValuesParser.unknownWord ([...]/stylelint-value-no-unknown-custom-properties/node_modules/postcss-values-parser/lib/ValuesParser.js:195:15)
    at ValuesParser.other ([...]/stylelint-value-no-unknown-custom-properties/node_modules/postcss-values-parser/lib/ValuesParser.js:110:10)
    at ValuesParser.parse ([...]/stylelint-value-no-unknown-custom-properties/node_modules/postcss-values-parser/lib/ValuesParser.js:135:16)
    at parse ([...]/stylelint-value-no-unknown-custom-properties/node_modules/postcss-values-parser/lib/index.js:21:12)
    at Function.fromTokens ([...]/stylelint-value-no-unknown-custom-properties/node_modules/postcss-values-parser/lib/nodes/Func.js:177:20)
    at ValuesParser.unknownWord ([...]/stylelint-value-no-unknown-custom-properties/node_modules/postcss-values-parser/lib/ValuesParser.js:152:12)
    at ValuesParser.other ([...]/stylelint-value-no-unknown-custom-properties/node_modules/postcss-values-parser/lib/ValuesParser.js:110:10)
    at ValuesParser.parse ([...]/stylelint-value-no-unknown-custom-properties/node_modules/postcss-values-parser/lib/ValuesParser.js:135:16) {
  reason: 'Unknown word',
  source: '--color, #{$color}',
  line: 1,
  column: 10,
  input: { line: 1, column: 10, source: '--color, #{$color}' }
}
```
</details>

**Comment by romainmenke** [comment]
SCSS isn't CSS :)

This plugin only works for CSS, anything else is beyond the scope of this plugin.
